src/main/java/websocket/WSManager.java

```diff
@@ -236,6 +242,14 @@ public void handleMessageSendError(String message) {
     public void setAuthInfo(String userID, String userToken) {
         this.userID = userID;
         this.userToken = userToken;
+        this.sendAllAuthenticatedRequests();
+    }
+
+    // TODO: fix potential concurrency issue with the list?
```

Comment:
This should be fixed before merging it in. A simple synchronized block around the insert and sendAllAuthenticatedRequests should suffice (if there are no other places that use it)